packages/opentelemetry-core/src/default_span.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Span,
+  SpanContext,
+  Status,
+  Attributes,
+  SpanKind,
+} from '@opentelemetry/types';
+
+/** This is a no-op implementation of Span. */
+export class DefaultSpan implements Span {
```

Comment:
`DefaultSpan` doesn't imply that it's no-op which could be confusing.

Reply:
This name (terminology) is aligned to across language specs. Do you have better options?

Reply:
IMHO the correct name is in the description. If this is a no-op implementation of a `Span` then it should be called `NoopSpan`. Then reading the description is not necessary as the behavior (or absence of behavior here) can be inferred from the name.

It also depends if this will be the base class for any other `Span` implementation and just happens to not do anything on its own. If that's the case, then it could also just be called `Span`.

Reply:
I am fine with either `BaseSpan` or `Span`.

Reply:
I am fine with these 2 as well if it is indeed the base span.

Reply:
> It also depends if this will be the base class for any other Span implementation and just happens to not do anything on its own.

@mayurkale22 will this be the intention moving forward? If so, I'm in favor of just `Span`.

Reply:
I thought over it again and decided to rename to `NoopSpan` instead of `BaseSpan` (this is not the base class for any other `Span` implementation).  Actually, this is the default span that is used when no `Span` implementation is available. All operations are no-op except context, also decided to return empty invalid span context instead of `null`. Please review again and let me know the thoughts?

Reply:
I think it makes sense that it's not the same since you shouldn't be expected to call the parent constructor. However, I still think there should be a real `SpanContext` for propagation purpose when the trace is dropped.